policy/approval/approve.go

```diff
@@ -133,19 +133,29 @@ func (r *Rule) Evaluate(ctx context.Context, prctx pull.Context) (res common.Res
 		}
 	}
 	res.PredicateResults = predicateResults
-	approved, msg, err := r.IsApproved(ctx, prctx)
+
+	allowedCandidates, err := r.FilteredCandidates(ctx, prctx)
+	if err != nil {
+		res.Error = errors.Wrap(err, "failed to filter candidates")
+		return
+	}
+
+	approved, msg, err := r.IsApproved(ctx, prctx, allowedCandidates)
```

Comment:
i left this bit because i still thought it was a good idea for refactoring the tests, but let me know if you want me to revert this too.